trunk/source/UltimateRegressionTest/src/de/uni_freiburg/informatik/ultimate/regressiontest/AbstractRegressionTestSuite.java

```diff
@@ -110,17 +117,82 @@ public Collection<UltimateTestCase> createTestCases() {
 		for (final Config runConfiguration : runConfigurations) {
 			final Collection<File> inputFiles = getInputFiles(filesRegexFilter, runConfiguration);
-
+			final NestedMap3<String, String, String, String> skippedTests = getSkippedTests(runConfiguration);
 			for (final File inputFile : inputFiles) {
 				final UltimateRunDefinition urd =
 						new UltimateRunDefinition(inputFile, runConfiguration.getSettingsFile(),
 								runConfiguration.getToolchainFile(), getTimeout(runConfiguration, inputFile));
-				rtr.add(buildTestCase(urd, getTestResultDecider(urd)));
+				final String overridenVerdict = skippedTests.get(inputFile.getName(),
+						runConfiguration.getSettingsFile().getName(), runConfiguration.getToolchainFile().getName());
+				rtr.add(buildTestCase(urd, getTestResultDecider(urd, overridenVerdict)));
 			}
 		}
 		return rtr;
 	}
 
+	private static NestedMap3<String, String, String, String> getSkippedTests(final Config runConfiguration) {
+		final NestedMap3<String, String, String, String> result = new NestedMap3<>();
+		final File settingsDir = runConfiguration.getSettingsFile().getParentFile();
+		final File toolchainDir = runConfiguration.getToolchainFile().getParentFile();
+		if (settingsDir.equals(toolchainDir)) {
+			addSkippedTest(new File(settingsDir, SKIPPED_FILENAME), result);
+		} else if (settingsDir.toString().length() < toolchainDir.toString().length()) {
+			addSkippedTest(new File(settingsDir, SKIPPED_FILENAME), result);
+			addSkippedTest(new File(toolchainDir, SKIPPED_FILENAME), result);
+		} else {
+			addSkippedTest(new File(toolchainDir, SKIPPED_FILENAME), result);
+			addSkippedTest(new File(settingsDir, SKIPPED_FILENAME), result);
+		}
+		return result;
+	}
+
+	private static void addSkippedTest(final File ignoreFile, final NestedMap3<String, String, String, String> map) {
+		try {
+			final Map<String, List<String>> parsed = new Yaml().load(new FileInputStream(ignoreFile));
+			for (final var entry : parsed.entrySet()) {
+				for (final String line : entry.getValue()) {
+					addSkipLine(line, entry.getKey(), map);
+				}
+			}
+		} catch (final FileNotFoundException e) {
+			// File does not exist, nothing to be ignored
+		}
+	}
+
+	private static void addSkipLine(final String line, final String exptectedVerdict,
+			final NestedMap3<String, String, String, String> map) {
+		if (line.startsWith("//")) {
+			return;
+		}
+		String settings = null;
+		String toolchain = null;
+		String file = null;
+		final String[] args = line.split("\\s+");
+		if (args.length != 3) {
+			return;
+		}
+		for (int i = 0; i < 3; i++) {
+			final String arg = args[i];
+			if (arg.endsWith(".epf")) {
+				if (settings != null) {
+					return;
+				}
+				settings = arg;
+			} else if (arg.endsWith(".xml")) {
+				if (toolchain != null) {
+					return;
+				}
+				toolchain = arg;
+			} else {
+				if (file != null) {
+					return;
+				}
+				file = arg;
+			}
+		}

```

Comment:
see first comment above: this can be simplified if we just use a YAML dictionary instead of encoding everything in one line